dev/hcloud-k8s-env/main-setup.tf

```diff
@@ -0,0 +1,216 @@
+# Setup the k3s cluster
+
+locals {
+  # The CIDR range for the Pods, must be included in the range of the
+  # network (10.0.0.0/8) but must not overlap with the Subnet (10.0.0.0/24)
+  cluster_cidr = "10.244.0.0/16"
+
+  registry_service_ip = "10.43.0.2"
+  registry_port       = 30666
+
+  kubeconfig_path = abspath("${path.root}/files/kubeconfig.yaml")
+  env_path        = abspath("${path.root}/files/env.sh")
+}
+
+resource "null_resource" "k3sup_control" {
+  triggers = {
+    id = hcloud_server.control.id
+    ip = hcloud_server_network.control.ip
+  }
+
+  connection {
+    host        = hcloud_server.control.ipv4_address
+    private_key = tls_private_key.ssh.private_key_openssh
+  }
+
+  provisioner "remote-exec" {
+    inline = ["mkdir -p /etc/rancher/k3s"]
+  }
+  provisioner "file" {
+    content = yamlencode({
+      "mirrors" : {
+        "localhost:${local.registry_port}" : {
+          "endpoint" : ["http://${local.registry_service_ip}:5000"]
+        }
+      }
+    })
+    destination = "/etc/rancher/k3s/registries.yaml"
+  }
+
+  provisioner "local-exec" {
+    command = <<-EOT
+      k3sup install --print-config=false \
+        --ssh-key='${local_sensitive_file.ssh.filename}' \
+        --ip='${hcloud_server.control.ipv4_address}' \
+        --k3s-channel='${var.k3s_channel}' \
+        --k3s-extra-args="\
+          --kubelet-arg=cloud-provider=external \
+          --cluster-cidr='${local.cluster_cidr}' \
+          --disable-cloud-controller \
+          --disable-network-policy \
+          --disable=local-storage \
+          --disable=servicelb \
+          --disable=traefik \
+          --flannel-backend=none \
+          --node-external-ip='${hcloud_server.control.ipv4_address}' \
+          --node-ip='${hcloud_server_network.control.ip}'" \
+        --local-path='${local.kubeconfig_path}'
+    EOT
+  }
+}
+
+resource "null_resource" "k3sup_worker" {
```

Comment:
Was the `depends_on` on Cilium+HCCM Helm Charts no longer necessary?

Reply:
Yes, I tested without it, and the whole problem was related to the nodes hostname.